components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -173,11 +287,82 @@ where
         let (resp, _) = request_with_retries.await?;
 
         let resp: Vec<ItemCredentialsResponse> = serde_json::from_slice(resp.body())?;
+        event!(Level::DEBUG, "done decoded body");
 
         let mut time_limited_creds: HashMap<String, Vec<TimeLimitedCredential>> = HashMap::new();
+        let mut time_limited_v2_creds: Vec::<String> = Vec::new();
 
         for item_cred in resp {
             match item_cred {
+                ItemCredentialsResponse::TimeLimitedV2 {
+                    item_id,
+                    blinded_creds,
+                    signed_creds,
+                    batch_proof,
+                    public_key,
+                    issuer_id,
+                    valid_from,
+                    valid_to,
+                    ..
+                } => {
+                    if let Some(item_creds) =
+                        self.client.get_time_limited_v2_creds(&item_id).await?
+                    {
+                        let from = NaiveDateTime::parse_from_str(&valid_from, "%Y-%m-%dT%H:%M:%SZ")
+                                    .map_err(|_| {
+                                        InternalError::InvalidResponse(
+                                            "Could not parse valid from".to_string(),
+                                        )
+                                    })?;
+                        let to = NaiveDateTime::parse_from_str(&valid_to, "%Y-%m-%dT%H:%M:%SZ")
+                                    .map_err(|_| {
+                                        InternalError::InvalidResponse(
+                                            "Could not parse valid to".to_string(),
+                                        )
+                                    })?;
+
+                        if item_creds.unblinded_creds.expect("stored creds missing").iter().find(|stored_cred| to == (*stored_cred).valid_to && from == (*stored_cred).valid_from).is_some() {
```

Comment:
probably should return an error if there are no unblinded creds instead of panicing?